plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/BaseMySqlConnectorTest.java

```diff
@@ -337,6 +342,87 @@ public void testPredicatePushdown()
                 .isFullyPushedDown();
     }
 
+    @Test
+    public void testPredicatePushdownWithCollation()
+    {
+        onRemoteDatabase().execute("CREATE TABLE tpch.nation_collate AS SELECT regionkey, nationkey, CONVERT(name USING utf8) COLLATE utf8_bin AS name FROM tpch.nation");
+
+        // varchar inequality
+        assertThat(query("SELECT regionkey, nationkey, name FROM nation_collate WHERE name != 'ROMANIA' AND name != 'ALGERIA'"))
+                .isFullyPushedDown();
```

Comment:
assert results

Reply:
we have `hasCorrectResultsRegardlessOfPushdown();` inside so it checks that results are the same with and without pushdown